pkg/generators/matrix.go

```diff
@@ -0,0 +1,82 @@
+package generators
+
+import (
+	"errors"
+	"time"
+
+	argoprojiov1alpha1 "github.com/argoproj-labs/applicationset/api/v1alpha1"
+	"github.com/argoproj-labs/applicationset/pkg/utils"
+)
+
+var _ Generator = (*MatrixGenerator)(nil)
+
+var MoreThanTwoGenerators = errors.New("found more than two generators, Matrix support only two")
+var LessThanTwoGenerators = errors.New("found less than two generators, Matrix support only two")
+var MoreThenOneInnerGenerators = errors.New("found more than one generator in matrix.Generators")
+
+type MatrixGenerator struct {
+	generators map[string]Generator
```

Comment:
I wasn't sure what this field was until I read through the code, what if we change the name of the  'generators' field to `supportedGenerators` with a comment like 'these are interfaces of all the other types of generators supported (cluster, list, git)'